trezorlib/tx_api.py

```diff
@@ -104,6 +109,30 @@ def json_to_tx(coin, data):
             extra_data_len = 1 + joinsplit_cnt * 1802 + 32 + 64
             t.extra_data = rawtx[-extra_data_len:]
 
+    if is_dash(coin):
+        t.type = data["type"]
+        if t.version == 3 and t.type != 1:
+            # It's a DIP2 special TX with payload
+            t.extra_data_len = data["extraPayloadSize"]
+            if t.extra_data_len == 0:
+                t.extra_data_len = 1
+                t.extra_data = b"\x00"
+            elif t.extra_data_len < 253:
```

Comment:
please extract this code into a separate function - something like `pack_varint(num)` which returns a `bytes` object

(if you could then use that function also in the zcash code, that would be extra nice .) )

alternately, if viable (how likely is it that the payload is more than 253 bytes?), you could be lazy, do the thing that ZCash does and raise if length is 253 or more

Reply:
It's actually a viable scenario that we get >= 253 bytes for the payload. Depends on which DIP2 types we later introduce. I'll extract the serialization code into its own method.

Reply:
I tried to use the freshly introduces `pack_varint` for ZCash as well, but then decided to not touch that part as it's not a simply replacement anymore. I would also have to change the way the data is extracted from `rawtx` and even though this is a small change, I find it to risky to touch code that I can't test/verify to work afterwards.